mlflow/gateway/providers/openai.py

```diff
@@ -0,0 +1,151 @@
+import openai
+from fastapi.encoders import jsonable_encoder
+
+from .base import BaseProvider
+from ..schemas import chat, completions, embeddings
+
+
+class OpenAIProvider(BaseProvider):
+    NAME = "openai"
+    SUPPORTED_ROUTES = ("chat", "completions", "embeddings")
+
+    async def chat(self, payload: chat.RequestPayload) -> chat.ResponsePayload:
+        resp = await openai.ChatCompletion.acreate(
+            model=self.config.model.name,
+            messages=jsonable_encoder(payload.messages),
+        )
+        # Response example (https://platform.openai.com/docs/api-reference/chat/create)
+        # ```
+        # {
+        #    "id":"chatcmpl-abc123",
+        #    "object":"chat.completion",
+        #    "created":1677858242,
+        #    "model":"gpt-3.5-turbo-0301",
+        #    "usage":{
+        #       "prompt_tokens":13,
+        #       "completion_tokens":7,
+        #       "total_tokens":20
+        #    },
+        #    "choices":[
+        #       {
+        #          "message":{
+        #             "role":"assistant",
+        #             "content":"\n\nThis is a test!"
+        #          },
+        #          "finish_reason":"stop",
+        #          "index":0
+        #       }
+        #    ]
+        # }
+        # ```
+        return chat.ResponsePayload(
+            **{
+                "candidates": [
+                    {
+                        "message": {
+                            "role": c["message"]["role"],
+                            "content": c["message"]["content"],
+                        },
+                        "metadata": {
+                            "finish_reason": c["finish_reason"],
+                        },
+                    }
+                    for c in resp["choices"]
+                ],
+                "metadata": {
+                    "input_tokens": resp["usage"]["prompt_tokens"],
+                    "output_tokens": resp["usage"]["completion_tokens"],
+                    "total_tokens": resp["usage"]["total_tokens"],
+                    "model": resp["model"],
+                    "route_type": self.config.type,
+                },
+            }
+        )
+
+    async def completions(self, payload: completions.RequestPayload) -> completions.ResponsePayload:
+        resp = await openai.Completion.acreate(
+            model=self.config.model.name,
+            prompt=payload.prompt,
```

Comment:
```suggestion
            prompt=payload.prompt,
            ...
```

Same here: https://github.com/mlflow/mlflow/pull/8697/files#r1227788703